lib/src/components/NxFileUpload/NxMultiFileUpload/NxMultiFileUpload.tsx

```diff
@@ -0,0 +1,180 @@
+  }
+
+  useEffect(function() {
+    if (fileRef.current) {
+      fileRef.current.scrollIntoView({block: 'nearest'});
+    }
+  }, []);
+
+  return (
+    <SelectedFile file={file} onDismiss={onDismiss} ref={fileRef}/>
+  );
+}
+
+const NxMultiFileUpload = forwardRef<HTMLDivElement, Props>(function NxMultiFileUpload(props, ref) {
+  const {
+        className: classNameProp,
+        onChange: onChangeProp,
+        files,
+        isRequired,
+        isPristine,
+        id,
+        disabled,
+        ...attrs
+      } = props,
+      isFileSelected = !!files?.length,
+      { showValidationErrors: formShowValidationErrors } = useContext(FormAriaContext),
+      showValidationErrors = formShowValidationErrors || !isPristine,
+      showError = isRequired && showValidationErrors && !isFileSelected,
+      className = classnames('nx-multi-file-upload', classNameProp),
+      noFileMessageClassName = classnames('nx-multi-file-upload__no-file-message', {
+        'nx-multi-file-upload__no-file-message--invalid': showError
+      }),
+      inputRef = useRef<HTMLInputElement>(null),
+      inputId = useUniqueId('nx-multi-file-upload-input', id),
+      selectedFilesContainerRef = useRef<HTMLDivElement>(null),
+      validationErrorId = useUniqueId('nx-multi-file-upload-validation-error');
+
+  function onChange(evt: FormEvent<HTMLInputElement>) {
+    const dataTransferObject = new DataTransfer();
+
+    // get the previous files uploaded
+    if (files) {
+      forEach((f) => dataTransferObject.items.add(f), Array.from(files));
+    }
+
+    const inputFiles = evt.currentTarget.files,
+        normalizedFiles = !inputFiles?.length ? null : inputFiles;
+
+    if (normalizedFiles) {
+      forEach((f) => dataTransferObject.items.add(f), Array.from(normalizedFiles));
+    }
+
+    onChangeProp(dataTransferObject.files);
+  }
+
+  function openPicker() {
+    inputRef.current?.click();
+  }
+
+  function onDismiss(fileObj: File) {
+    const closeBtns = Array.from(selectedFilesContainerRef.current?.
+        querySelectorAll<HTMLButtonElement>('.nx-selected-file__dismiss-btn') ?? []);
+
+    if (files) {
+      const filesArray = Array.from(files);
+      filesArray.map((f) => {
+        if (f === fileObj) {
+          const idx = filesArray.indexOf(f);
+          if (idx === files.length - 1) {
+            // if the selected file is the last in the container, set the focus to the next last dismiss button
+            // if selected file is only file, set focus on the input
+            closeBtns.length === 1 ? inputRef.current?.focus() : closeBtns[idx - 1].focus();
+          }
+          else {
+            // need to explicity set focus on next closeBtn for Safari
+            closeBtns[idx].focus();
+          }
+        }
+      });
+    }
+
+    const dataTransferObject = new DataTransfer();
+    if (inputRef.current) {
+      if (files) {
+        forEach((f: File) => dataTransferObject.items.add(f), without([fileObj], Array.from(files)));
```

Comment:
How about a utility function that copies combines one or more `FileList`s (passed as varargs) and returns the new one.  That would encapsulate all the `DataTransfer` stuff that's currently spread across two places

Reply:
> How about a utility function that copies/combines one or more `FileList`s (passed as var-args) and returns the new one. That would encapsulate all the `DataTransfer` stuff that's currently spread across two places

Created a function here: [8dbfe33](https://github.com/sonatype/sonatype-react-shared-components/pull/850/commits/8dbfe33b086bab83f311ec0e334985de01af8313)